src/three-domain-secure/component.jsx

```diff
@@ -94,6 +94,13 @@ export function getThreeDomainSecureComponent(): TDSComponent {
           queryParam: "cart_id",
           // $FlowFixMe[incompatible-call]
           queryValue: ({ value }) => ZalgoPromise.try(value),
+          required: false,
+        },
+        vaultToken: {
+          type: "string",
+          queryParam: "token",
+          queryValue: ({ value }) => value,
```

Comment:
I think this should be `value`

Reply:
oh actually I don't think we need this line at all:
```
  if (value === null || typeof value === 'undefined' || !propDef) {
    return;
  }

  ...
  
  finalValue: ZalgoPromise.try(() => {
    if (typeof queryValue === 'function' && isDefined(value)) {
      return queryValue({
        value
      });
    } else {
      return value;
    }
  })
```

If there is no value, the function early returns and doesn't include the prop as a query param.

If there is a value, it is used as-is.

`queryValue` is for modifying the value before sending it as a query param